index.js

```diff
@@ -325,6 +325,20 @@ function iterableEqual(leftHandOperand, rightHandOperand, options) {
       return false;
     }
   }
+
+  var leftHandProps = Object.keys(leftHandOperand);
+  var rightHandProps = Object.keys(rightHandOperand);
+  if (leftHandProps.length !== rightHandProps.length) {
+    return false;
+  }
+
+  for (var prop in rightHandProps) {
```

Comment:
Using `for ... in` the way it is used here would actually open up another bug as it would also test for inherited properties.
The fasted way to check this properly is to use `Object.keys` with a plain for loop.